core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -218,41 +219,125 @@ class SnappyContext protected[spark](
    * @todo provide lot more details and examples to explain creating and
    *       using sample tables with time series and otherwise
    * @param tableName the qualified name of the table
+   * @param samplingOptions
+    */
+  def createSampleTable(tableName: String, samplingOptions: Map[String, String]): DataFrame = {
+    val plan = createTable(catalog.newQualifiedTableName(tableName),
+      SnappyContext.SAMPLE_SOURCE, None, schemaDDL = None,
+      SaveMode.ErrorIfExists, samplingOptions,
+      onlyBuiltIn = true, onlyExternal = false)
+    DataFrame(self, plan)
+  }
+
+  /**
+   * Create a stratified sample table. Java friendly version.
+   * @todo provide lot more details and examples to explain creating and
+   *       using sample tables with time series and otherwise
+   * @param tableName the qualified name of the table
+   * @param samplingOptions
+   */
+  def createSampleTable(tableName: String
+      , samplingOptions: java.util.Map[String, String]): DataFrame = {
+    createSampleTable(tableName, samplingOptions.asScala.toMap)
+  }
+
+
+  /**
+   * Create a stratified sample table.
+   * @todo provide lot more details and examples to explain creating and
+   *       using sample tables with time series and otherwise
+   * @param tableName the qualified name of the table
    * @param schema
    * @param samplingOptions
-   * @param ifExists
    */
-  def createSampleTable(tableName: String, schema: Option[StructType],
-      samplingOptions: Map[String, String],
-      ifExists: Boolean = false): DataFrame = {
+  def createSampleTable(tableName: String, schema: StructType,
+      samplingOptions: Map[String, String]): DataFrame = {
     val plan = createTable(catalog.newQualifiedTableName(tableName),
```

Comment:
@rishitesh

That's an oversight because DDL does allow for it (in Spark SQL too), so we should also add to createTable.
Dropping the table and recreating is no solution since the data will be lost. One can instead catch the exception and ignore in code but that is ugly. As I said DDL does allow for it so the API should also allow.